### PR TITLE
Extension Fire button

### PR DESCRIPTION
**Pull request by sammacbeth**
**Reviewer:**

## Description:

- Adds an extension fire button popup to the dashboard
- This is only shown when specifically asked for via an option in the `getPrivacyDashboardData` message.

### CODE REVIEW

File: integration-tests/browser.spec-int.js [hunk 1]
```diff
@@ -142,6 +142,121 @@ test.describe('localization', () => {
...
+        const buttons = await dash.fireButton().all()
+        expect(buttons).toHaveLength(0)
+    })
+
+    test('adding firebutton option to dashboard message shows fire button', async ({ page }) => {
+        const mock = new MockData({ url: 'https://example.com' })
+        const messages = mockToExtensionDashboardMessage(mock, true)
+        const dash = await DashboardPage.browser(page, messages)
+        await dash.fireButton().waitFor()
+        expect(await dash.fireButton().isVisible()).toBeTruthy()
```
Comment (shakyShane) [inline integration-tests/browser.spec-int.js:160]: can you add a screenshot for this state (see the bottom of this file). Even if we just add one for the primary screen when the fire button is showing that would be good and prevent others from breaking it :)

**Review by shakyShane — CHANGES_REQUESTED** [review]

File: shared/js/ui/views/tests/generate-data.mjs [hunk 2]
```diff
@@ -294,6 +294,9 @@ export function mockToExtensionDashboardMessage(mock) {
         },
         requestData: { requests: mock.requests },
         emailProtectionUserData: undefined,
+        fireButton: {
+            enabled: withFireButton,
```
Comment (shakyShane) [inline shared/js/ui/views/tests/generate-data.mjs:298]: I noticed on the preview page that we cannot see the variant without the fire button - I think we should add another iframe and ensure we can review both side-by-side

**Review by shakyShane — CHANGES_REQUESTED** [review]

File: shared/js/ui/views/tests/generate-data.mjs (hunk 2 above)
Comment (shakyShane) [reply shared/js/ui/views/tests/generate-data.mjs:298]: Screenshot 2023-06-30 at 09 21 00

File: integration-tests/helpers.js [hunk 3]
```diff
@@ -235,11 +235,19 @@ export async function installBrowserMocks(page) {
...
         }
         try {
             if (!window.chrome) {
                 // @ts-ignore
-                window.chrome = {}
+                window.chrome = {
+                    // @ts-ignore
+                    permissions: {
+                        // eslint-disable-next-line n/no-callback-literal
+                        request: (permissions, cb) => cb && cb(true),
```
[the remark this answers is not in the document]
Comment (sammacbeth) [reply integration-tests/helpers.js:248]: No. In Chrome, the permission we're asking for (`browsingData`) is automatically granted when the API is called after a user action. Therefore, we do not expect this to ever be declined.

File: schema/fire-button.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                        "openTabs": {
+                            "type": "number"
+                        },
+                        "cookies": {
+                            "type": "number"
+                        },
+                        "pinnedTabs": {
+                            "type": "number"
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (sammacbeth) [reply schema/fire-button.json:66]: Added a list of required fields to the definition.

File: schema/fire-button.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    "enum": ["CurrentSite", "LastHour", "Last24Hour", "Last7days", "Last4Weeks", "AllTime"]
+                },
+                "selected": {
+                    "type": "boolean"
+                },
+                "options": {
+                    "type": "object",
+                    "additionalProperties": true
+                },
+                "descriptionStats": {
```
[the remark this answers is not in the document]
Comment (sammacbeth) [reply schema/fire-button.json]: I forgot to add these types. Added now.

File: integration-tests/browser.spec-int.js [hunk 6]
```diff
@@ -240,20 +215,10 @@ test.describe('fire button', () => {
...
-                {
-                    messageType: 'doBurn',
-                    options: getBurnOptions.options[1].options,
-                },
-            ],
-        ])
+        await dash.clickFireButton()
+        await dash.chooseBurnOption(1)
+        await dash.clickFireButtonBurn()
+        await dash.sendsOptionsWithBurnMessage(getBurnOptions.options[1].options)
```
Comment (shakyShane) [inline integration-tests/browser.spec-int.js]: :chefs_kiss: ❤️

File: schema/fire-button.json [hunk 7]
```diff
@@ -30,6 +30,7 @@
                 },
                 "descriptionStats": {
                     "type": "object",
+                    "required": ["openTabs", "cookies", "pinnedTabs", "clearHistory", "duration"],
```
Comment (shakyShane) [inline schema/fire-button.json:44]: nice!

File: shared/js/ui/views/tests/generate-data.mjs [hunk 8]
```diff
@@ -543,10 +553,24 @@ export const createDataStates = (google, cnn) => {
...
+        'fire-button': new MockData({
+            requests: google.requests,
+            url: 'https://google.com',
+            parentEntity: {
+                displayName: 'Google',
+                prevalence: 80.1,
+            },
+            fireButtonEnabled: true,
+            fireButtonOptions: { clearHistory: true, tabClearEnabled: true, pinnedTabs: 2 },
+        }),
```
Comment (shakyShane) [inline shared/js/ui/views/tests/generate-data.mjs:565]: 👌🏻

**Review by shakyShane — APPROVED** [review]
Amazing - thanks for making those changes :)

**Pull request merged**